submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionRequestMongoPersistenceService.kt

```diff
@@ -63,17 +61,9 @@ class SubmissionRequestMongoPersistenceService(
         return requestRepository.getByAccNoAndVersion(accNo, version).status
     }
 
-    override suspend fun updateRqtFile(
-        rqt: SubmissionRequestFile,
-        options: UpdateOptions?,
-    ) {
-        val accNo = rqt.accNo
-        val version = rqt.version
-        requestRepository.updateIndex(accNo, version, rqt.index)
-        when (options) {
-            UPDATE_FILE -> requestRepository.updateSubRqtFile(accNo, version, rqt.path, rqt.file)
-            null -> {}
-        }
+    override suspend fun updateRqtFile(rqt: SubmissionRequestFile) {
+        requestRepository.updateIndex(rqt.accNo, rqt.version)
```

Comment:
Not sure if I'm being overly-complicated here but, perhaps we should increase the index once the update operation is finished and not before i.e. swap the lines